packages/api/src/providers/oauth1.ts

```diff
@@ -161,23 +161,39 @@ export class OAuth1DefaultImpl implements OAuth1 {
     return { oauth_token, oauth_token_secret };
   }
 
-  async deregister(userAccessTokens: string[]): Promise<void> {
+  /**
+   * Deregisters all users with the given UATs. Could be used to revoke tokens from all customers or a specific one
+   *
+   * @param {string[]} userAccessTokens list of token strings
+   * @param {string} cxId customer ID to revoke the token for a specific customer only
+   */
+  async deregister(userAccessTokens: string[], cxId?: string): Promise<void> {
     for (const oauthUserAccessToken of userAccessTokens) {
       const userTokenList = await getUserTokenByUAT({ oauthUserAccessToken });
       for (const userToken of userTokenList) {
-        capture.setUser({ id: userToken.userId });
         // DynamoDB (Webhook and auth)
-        const updatedUserToken = userToken.clone();
-        updatedUserToken.oauthUserAccessToken = undefined;
-        updatedUserToken.oauthUserAccessSecret = undefined;
-        await saveUserToken(updatedUserToken);
-        // Postgres (app standard)
-        await updateProviderData({
-          id: userToken.userId,
-          cxId: userToken.cxId,
-          provider: this.providerName,
-          providerItem: undefined,
-        });
+        try {
+          if (!cxId || cxId === userToken.cxId) {
+            capture.setUser({ id: userToken.cxId });
+            const updatedUserToken = userToken.clone();
+            updatedUserToken.oauthUserAccessToken = undefined;
+            updatedUserToken.oauthUserAccessSecret = undefined;
+            await saveUserToken(updatedUserToken);
+
+            // Postgres (app standard)
+            await updateProviderData({
+              id: userToken.userId,
+              cxId: userToken.cxId,
+              provider: this.providerName,
+              providerItem: undefined,
+            });
+          }
+        } catch (error) {
+          console.log("OAuth1 deregester failed", error);
+          capture.error("OAuth1 deregister failed", {
```

Comment:
`capture.error` is supposed to be called with the instance of `Error` as the first parameter. To "capture" with text we can use `capture.message`.

The compiler doesn't complain b/c `error` can be a string or an Error, so its treated as `unknown`, and we built `capture.error` accepting `any` so we don't have to narrow down the type of `error` on `catch` every time we want to call `capture.error`... it would need to be something like this:

```ts
...
} catch (error) {
  if (error instanceof Error) capture.error(error, ...);
  else capture.message(error, ...);
}
```